### ISSUE TITLE
No error for YAML aliases, which CloudFormation rejects ("YAML aliases are not allowed")

### ISSUE THREAD

**Issue by esetnik** [aws-cloudformation/cfn-lint#4721]
### CloudFormation Lint Version

1.57.0 (also reproduced on 1.48.1)

### What operating system are you using?

Mac

### Describe the bug

cfn-lint reports nothing for a template that uses YAML anchors and aliases, but CloudFormation rejects any such template:

```
$ cfn-lint template.yaml; echo $?
0

$ aws cloudformation validate-template --template-body file://template.yaml
An error occurred (ValidationError) when calling the ValidateTemplate operation: Template error: YAML aliases are not allowed in CloudFormation templates
```

`aws cloudformation deploy` fails the same way at `CreateChangeSet`. (AWS CLI 2.36.15, us-east-1.)

So a template that lints clean can still fail on its first deploy. We only found out when the change set was rejected.

### Expected behavior

An error (or at least a warning) at the alias. For example: "YAML aliases are not allowed in CloudFormation templates; expand the value or process the template with `aws cloudformation package` first."

Related but different: #2993 covers anchors combined with merge keys, where the commenter notes that `aws cloudformation package` expands aliases before deploy. That route is valid, so a warning that mentions it may fit better than a hard error. The failure is for templates submitted directly: ValidateTemplate, CreateStack, CreateChangeSet, and `aws cloudformation deploy` without `package`.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Description: YAML alias reuse that CloudFormation rejects at CreateChangeSet
Resources:
  FirstLogGroup:
    Type: AWS::Logs::LogGroup
    Properties:
      RetentionInDays: &Retention 30
  SecondLogGroup:
    Type: AWS::Logs::LogGroup
    Properties:
      RetentionInDays: *Retention
```

**Comment by esetnik** [comment]
I'd like to fix this and have a PR ready if the approach sounds OK. The existing alias warning, W1101, finds aliases by object identity: it reports a mapping or sequence that is reachable from two places. That can't see scalar aliases like `*Retention` above, because an aliased `30` is indistinguishable from any other `30` (CPython shares small ints, `True`, `False`, etc.). Proposal: keep resolving aliases exactly as today, but have `cfn_yaml.loads` also record where each alias token appears. It would use a libyaml token scan that only runs when the source contains `&`, the same gate `_guard_alias_expansion` already uses. `Template` captures that list before transforms run, and W1101 reports one warning per alias at the alias's own line and column, whatever the aliased value is. W1101 stays a warning (resolution is deliberate for `aws cloudformation package`/SAM users), and it would skip templates that declare `Transform: AWS::Serverless-2016-10-31`, since those are normally deployed with the SAM CLI, which expands aliases client-side. Happy to adjust if you'd prefer a different shape.
